### ISSUE TITLE
Testing notes Oct 26, 2017

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#322]
Great work @jessegreenberg 
These are the things I noticed today.

**Reading Through:**
- [x] Punctuation: Sweater DD has 3 extra periods.
- [x] Hear Two-balloon experiment label twice - Possibly due to order, possibly normal repetition dues to label and for attribute (aria-label might be better for the checkbox since there is no text on screen?) See issue #330
- [x] Same label repetition for radios. Text on screen in this case, so I would stick with an explicit label. @terracoda to investigate repetition for label is used with for attribute. Fixed by system update. No longer hearing repetition on radios with High Sierra and Safari 11.0.1. (Checking off and starting new issue for this item).
- [x] No focus highlight after using Cntrl-Opt-B to read from the beginning of the sim.
- [x] Sometimes I don't get any alerts. It just seems that VO konks out, and a page reload is required.

**Interacting:**
- [x] Release alert: Use very simple induced charge pattern for releases that end at wall. Specifically, the Induced charge dragging (generic) pattern. Design doc updated. Making a new issue (#332) for this, and @terracoda to make sure this is clear in the design documents that describe releases. 
- [x] Punctuation: Grab Alert, missing 2 periods (End of Induced charge, End of Wall Charge).
- [x] Grab Alert missing balloon charge.
- [x] Regularly hear "Off sweater." When sim screen finishes loading. Could this be some stale alert content?
- [x] Some duplicate words in locations, e.g., heard "Near Near wall." When neutral balloon released near wall, and
- [x] in Grab Alert, heard: "At At center of Play Area."
- [x] When rubbing on Sweater. I never get the "few more" description. Could we make sure that fires at least once, instead of multiple "picks up more negative charges?"
- [x] Diagonal movement might need a lower threshold in Play Area. They work great for releases on Sweater, though! Started specific issue for this #331

**Complicated Release Path:**
- [x] Almost got funny movement described when Green Balloon moved right, then left, still moving left across bottom of sweater!!!! But it needs some tweaking. Didn't get final location when it stopped on Sweater. @terracoda to work in this in design document.

**Independent Balloon Movement**
- [x] Independent movement of un-grabbed balloon being tracked. Needs label and maybe some tweaking, but working!!!! @terracoda to work in this in design document.

**When using Mouse with VO On**
- [x] Multiple alerts string up and keep going and going and going (was using mouse). Not sure how to deal with that, but maybe alert queue is not set-up for Mouse interaction. I only notice this as I was quickly setting up a scenario with the mouse.

**Comment by jessegreenberg** [comment]
> Hear Two-balloon experiment label twice

I am not able to reproduce this one with NVDA or VoiceOver.

**Comment by jessegreenberg** [comment]
Ah, it happens when reading with the virtual cursor. Now I hear it. @terracoda said
> @terracoda to investigate repetition for label is used with for attribute.

The HTML that looks like
```html
<div tabindex="-1" id="container-62-125-359-435-426-409">
  <input id="62-125-359-435-426-409" type="checkbox">
  <label tabindex="-1" id="label-62-125-359-435-426-409" for="62-125-359-435-426-409">Two-Balloon Experiments</label>
  <p tabindex="-1" id="description-62-125-359-435-426-409">Play with two balloons or just one.</p>
</div>
```
It seems like the AT is reading the accessible name on the input when it receives focus,  then reading the content on  the  label element with the virtual cursor again. @terracoda can you verify?

**Comment by jessegreenberg** [comment]
>  No focus highlight after using Cntrl-Opt-B to read from the beginning of the sim.

Checking to see if this is a sim problem or a browser issue.

**Comment by jessegreenberg** [comment]
hmm, now focus is behaving just fine with the virtual cursor.

**Comment by jessegreenberg** [comment]
It seems to happen every time I start reading from the top (ctrl + option + b), if I start reading from below the balloon it doesn't happen (ctrl + shift + a). Checking events in the console.

**Comment by jessegreenberg** [comment]
I think VoiceOver is trying to send the focus event to things that are not "focusable", and since those node's don't have defined bounds, scenery doesn't know how to draw the focus highlight. After the error, it never recovers. I can see that whatever has focus will  blur when safari  hits the scene summary list.

**Comment by jessegreenberg** [comment]
Preventing scenery from focusing  Nodes that are not focusable seems to fix it. ctrl + option + b 5 times and haven't run into this problem. Checking off the item!

**Comment by jessegreenberg** [comment]
> Sometimes I don't get any alerts. It just seems that VO konks out, and a page reload is required

Unfortunately I haven't run into this yet. This sounds a lot like a performance problem.

**Comment by jessegreenberg** [comment]
The above commit prevents too many alerts related to direction from coming through, I think it sounds nice when using a mouse now. Thanks @terracoda, good idea.

**Comment by terracoda** [comment]
@jessegreenberg, regarding https://github.com/phetsims/balloons-and-static-electricity/issues/322#issuecomment-342329694, looking forward to testing this fix out. 

**Question:** Does the fix here, fix it for other sims as well?

I am asking because yesterday, I was working with screen reader consultant who likes to interact with the track pad and she said she was getting stuck in the live region, but I think what was happening was that alerts were queuing up and just being delivered one after the other until they were all done. That is what it sounded like to me.

**Comment by jessegreenberg** [comment]
@terracoda no, this was a sim specific fix for the balloon interaction. What sim was the consultant using?

**Comment by terracoda** [comment]
@jessegreenberg, sonified Ohm's Law. I have clean notes in the Sonfication Work/ Screen Reader Sessions, but I have not yet fully assessed the issues and created github issues.

**Comment by jessegreenberg** [comment]
@terracoda I think the remaining items for this issue are for you, but please reassign to me if there is more for me to work on for the remaining check boxes.

**Comment by terracoda** [comment]
@jessegreenberg for the remaining items, I have created new issues, so I will close this one, and assign new issues to you when I have the design documentation ready.
